plantcv/plantcv/photosynthesis/analyze_yii.py

```diff
@@ -0,0 +1,165 @@
+
+    # Calculate which non-zero bin has the maximum Fv/Fm value
+    max_bin = midpoints[np.argmax(yii_hist)]
+
+    # Create a dataframe
+    hist_df = pd.DataFrame({'Plant Pixels': yii_hist, measurement: midpoints})
+
+    # Make the histogram figure using plotnine
+    hist_fig = (ggplot(data=hist_df, mapping=aes(x=measurement, y='Plant Pixels'))
+                + geom_line(show_legend=True, color="green")
+                + geom_label(label=f"Peak Bin Value: {str(max_bin)}", x=.15, y=205, size=8, color="green"))
+
+    # Plot/Print out the induction curves and the histograms
+    _debug(visual=ind_fig, filename=os.path.join(params.debug_outdir, str(params.device) + "_fluor_induction.png"))
+    _debug(visual=hist_fig, filename=os.path.join(params.debug_outdir, str(params.device) + "_YII_histogram.png"))
+
+    # Store images
+    outputs.images.append(yii_img)
+
+    # Create variables to label traits as either Fv/Fm or Fv'/Fm' measurements
+    var = measurement.lower()
+    var = var.replace("'", "p")
+    var = var.replace("/", "")
+
+    outputs.add_observation(sample=label, variable=f"{var}_hist", trait=f"{measurement} frequencies",
+                            method='plantcv.plantcv.photosynthesis.analyze_fvfm', scale='none', datatype=list,
+                            value=yii_hist.tolist(), label=np.around(midpoints, decimals=len(str(bins))).tolist())
+    outputs.add_observation(sample=label, variable=f"{var}_hist_peak", trait=f"peak {measurement} value",
+                            method='plantcv.plantcv.photosynthesis.analyze_fvfm', scale='none', datatype=float,
+                            value=float(max_bin), label='none')
+    outputs.add_observation(sample=label, variable=f"{var}_median", trait=f"{measurement} median",
+                            method='plantcv.plantcv.photosynthesis.analyze_fvfm', scale='none', datatype=float,
+                            value=float(np.around(yii_median, decimals=4)), label='none')
+
+    return ind_fig, hist_fig, yii_img
+
+
+def _fluor_induction(ps, mask, measurement):
+    """Calculate fluorescence induction curve.
+
+    Inputs:
+    ps          = photosynthesis xarray DataArray
+    mask        = mask of plant (binary, single channel)
+    measurement = choose which measurement routines to analyze: light, dark, or both (default)
+
+    Returns:
+    df          = data frame of fluorescence in the masked region at each timepoint
+
+    :param ps: xarray.core.dataarray.DataArray
+    :param mask: numpy.ndarray
+    :param measurement: str
+    :return df: pandas.core.frame.DataFrame
+    """
+    # Prime is empty for Fv/Fm (dark- and light-adapted) and ' for Fq'/Fm'
+    prime = ""
+    if measurement.lower() == "fq'/fm'":
+        prime = "'"
+    # Create a list of fluorescence values, measurement labels, and timepoint indices
+    fluor_values = []
+    meas = []
+    timepts = []
+    # Iterate over the PAM measurement frames
+    for i in range(ps.attrs[f"F{prime}-frames"]):
```

Comment:
this would be more idiomatic
```
from sklearn.utils import img_as_bool
fluor_values = psii.where(np.expand_dims(img_as_bool(mask),2)).mean(axis=(0,1))
```
and 
`meas = psii.coords['frame_label']`

and then subset for F{\d+}

Reply:
but in the end i still had to loop to subset the fluor_values and meas labels
```
idx=[]
for i, s in enumerate(ps.coords['frame_label'].data):
    if re.search(r'\d+', s):
        idx.append(i)

idx        
fluor_values = psii.where(np.expand_dims(img_as_bool(mask),2)).mean(axis=(0,1))[idx]

meas = psii.coords['frame_label'][idx]

```

Reply:
oh! they have string methods. 
```
idx = [psii.coords['frame_label'].str.contains('\d+')]
meas = psii.coords['frame_label'][idx]
fluor_values = psii.where(np.expand_dims(img_as_bool(mask),2)).mean(axis=(0,1))[idx]